src/InstanceSerialization/InstanceMetadata.php

```diff
@@ -92,13 +92,22 @@ public function __construct(string $instance) {
 
       $this->as_float32[] = strpos($property->getDocComment(), '@kphp-serialized-float32') !== false;
 
+      // get type either from @var or from php 7.4 field type hint
+      $type = '';
       preg_match('/@var\s+([^\n]+)/', $property->getDocComment(), $matches);
-      assert(count($matches) > 1);
-      $type     = (string)$matches[1];
+      if (count($matches) > 1) {
+        $type = (string)$matches[1];
+      } else if (version_compare(phpversion(), '7.4', '>=') && $property->hasType()) {
```

Comment:
Can be simplified to `PHP_VERSION_ID >= 70400`.

Reply:
+